modules/data_encoding/message_encoding_decoding.py

```diff
@@ -0,0 +1,78 @@
+"""
+Encoding and Decoding Global Positions
+Save first byte as char to represent which worker sent the message 
+"""
+
+import struct
+
+from .worker_enum import WorkerEnum
+
+from .. import position_global
+
+
+# For Floats
+def encode_position_global(
+    worker_name: str, global_position: position_global.PositionGlobal
+) -> "tuple[True, bytes] | tuple[False, None]":
+    """
+    Encode PositionGlobal object into Bytes. Worker_ID to be encoded as the first byte of the message
+
+    Parameters:
+       worker_name: name of the worker defined by the name of its own file (worker_name = pathlib.Path(__file__).stem)
+       global_position: PositionGlobal object
+
+    Returns:
+        packed_coordinates (bytes): Encoded latitude, longitude, altitude of PositionGlobal object as bytes.
+        First byte dependant on which worker is calling the funciton, value depends on its corresponding enum value.
+    """
+
+    try:
+
+        worker_id = WorkerEnum[worker_name.upper()]
+        if not worker_id:
+            return False, None
+
+        # Encode message using PositionGlobal's latitude, longitude, altitude, with the worker ID in the front
+        packed_coordinates = struct.pack(
+            "=Bddd",
+            worker_id.value,  # 1 byte (char)
+            global_position.latitude,
+            global_position.longitude,
+            global_position.altitude,  # 3 double-precision floats
+        )
+    except (struct.error, AttributeError, ValueError):
+        return False, None
+
+    return True, packed_coordinates
+
+
+def decode_bytes_to_position_global(
+    encoded_global_position: bytes,
+) -> "tuple[True, int, position_global.PositionGlobal] | tuple[False, None, None]":
+    """
+    Decode bytes into a PositionGlobal object.
+
+    Parameters:
+        encoded_message (bytes): Encoded bytearray containing latitude, longitude, altitude
+
+    Returns:
+        Tuple: success, unsigned char: worker_id_number PositionGlobal: Decoded PositionGlobal object.
+    """
+
+    # check to make sure encoded message is 1 char and 3 double precision floats
+
+    # Unpack the byte sequence
+    try:
+        if len(encoded_global_position) != struct.calcsize(
+            "=Bddd"
```

Comment:
Does this require the equals sign? I feel like if you leave it as default like the other ones it should be fine

Reply:
I've found that basically yes it does. Without the equal sign struct seems to automatically add padding bytes. struct.calcsize("Bddd") is 32 while struct.calcsize("=Bdd") is 25

Reply:
as far as i know this is only the case if your grouping different types when packing